queries.sql

```diff
@@ -86,3 +86,70 @@ GROUP BY owners.full_name
 ORDER BY animal_count DESC
 LIMIT 1;
 
+-- Write queries to answer the following:
+-- Who was the last animal seen by William Tatcher?
+-- How many different animals did Stephanie Mendez see?
+-- List all vets and their specialties, including vets with no specialties.
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+-- What animal has the most visits to vets?
+-- Who was Maisy Smith's first visit?
+-- Details for most recent visit: animal information, vet information, and date of visit.
+-- How many visits were with a vet that did not specialize in that animal's species?
+-- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
+
+SELECT * FROM visits
+JOIN animals ON visits.animal_id = animals.id
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'William Tatcher'
+ORDER BY visits.date_of_visit DESC
+LIMIT 1;
+
+SELECT COUNT(DISTINCT animals.id) FROM visits
+JOIN animals ON visits.animal_id = animals.id
+JOIN owners ON visits.vet_id = owners.id
+WHERE owners.full_name = 'Stephanie Mendez';
+
+SELECT *
+FROM vets
+JOIN specializations ON vets.id = specializations.vet_id;
+
+SELECT animals.name FROM visits
+JOIN animals ON visits.animal_id = animals.id
+JOIN owners ON visits.vet_id = owners.id
+WHERE owners.full_name = 'Stephanie Mendez'
+AND visits.date_of_visit BETWEEN '2020-04-01' AND '2020-08-30';
+
+SELECT animals.name, COUNT(visits.id) AS visit_count FROM visits
+JOIN animals ON visits.animal_id = animals.id
+GROUP BY animals.name
+ORDER BY visit_count DESC
+LIMIT 1;
+
+SELECT * FROM visits
+JOIN animals ON visits.animal_id = animals.id
+JOIN vets ON visits.vet_id = vets.id
+JOIN owners ON visits.vet_id = owners.id
+WHERE owners.full_name = 'Maisy Smith'
+ORDER BY visits.date_of_visit ASC
+LIMIT 1;
+
+SELECT animals.name, vets.name, visits.date_of_visit FROM visits
+JOIN animals ON visits.animal_id = animals.id
+JOIN vets ON visits.vet_id = vets.id
+ORDER BY visits.date_of_visit DESC
+LIMIT 1;
+
+SELECT COUNT(visits.animal_id)
+FROM visits
+JOIN vets ON visits.vet_id = vets.id
+JOIN specializations ON specializations.species_id = vets.id
+WHERE specializations.species_id IS NULL;
+
+SELECT COUNT(visits.animal_id), animals.name
+FROM visits
+JOIN vets ON visits.vet_id = vets.id
+JOIN animals ON visits.animal_id = animals.id
+WHERE vets.name = 'Maisy Smith'
+GROUP by animals.name
+ORDER by COUNT(animals.name) DESC
+LIMIT 1 ;
```

Comment:
- [ ] As per the [required query](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_join_table.md#:~:text=What%20specialty%20should%20Maisy%20Smith%20consider%20getting%3F%20Look%20for%20the%20species%20she%20gets%20the%20most.), please select the specialty   'Maisy Smith' gets the most and not the animal name.